src/galaxy/StarSystem.cpp

```diff
@@ -186,6 +186,9 @@ const float StarSystem::starScale[] = {  // Used in sector view
 	4.0f  // Supermassive blackhole
 };
 
+int supportComplexLife;
+int LifeChanceMultiplier = (rand() % static_cast<int>(4)); //random value between 0 and 3
```

Comment:
Depends. Is the intention here that at equivalent oxidizing level all planet should support complex life the same way, or should there be some variation ? I'm enclined to have the latter. If so, then yes the variables should definitely be inside the function body, as otherwise all bodies would have the same multiplier.

But in any case, this should be computed only once during the body initialization, not in the getter.